app/controllers/posts_controller.rb

```diff
@@ -137,6 +133,11 @@ def authorize_developer
     redirect_to root_path, alert: 'You can only edit your own posts' unless editable?(@post)
   end
 
+  def no_self_like
+    post = Post.find_by(slug: params[:slug])
```

Comment:
what if you add a custom validation in the post like model (not sure the name), something like:

```ruby
validates :ensure_no_self_like

def ensure_no_self_like
  return true if owner != post.owner

  errors.add(:owner, 'You can you can not like your own post')
end
```

Reply:
the is no like model!  Thats why the validation is a little weird